packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -241,6 +227,66 @@ class ChromiumLauncher {
     ), skipCheck);
   }
 
+  Future<Process> _spawnChromiumProcess(List<String> args) async {
+    // Keep attempting to launch the browser until one of:
+    // - We successfully launched it, in which case we just return from the loop.
+    // - We encountered an unretriable error, in which case we throw ToolExit.
+    while (true) {
+      final Process process = await _processManager.start(args);
+
+      bool exited = false;
+      unawaited(process.exitCode.then((int exitCode) {
+        exited = true;
+      }));
+
+      process.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((String line) {
+          _logger.printTrace('[CHROME]: $line');
+        });
+
+      // Wait until the DevTools are listening before trying to connect. This is
+      // only required for flutter_test --platform=chrome and not flutter run.
+      //
+      // Once every few thousands of launches we hit this glibc bug: https://sourceware.org/bugzilla/show_bug.cgi?id=19329.
+      // When this happens Chrome spits out something like the following then exits with code 127:
+      //
+      //     Inconsistency detected by ld.so: ../elf/dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
+      bool hitGlibcBug = false;
+      await process.stderr
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .map((String line) {
+          _logger.printTrace('[CHROME]:$line');
+          if (line.contains('Inconsistency detected by ld.so')) {
+            hitGlibcBug = true;
+          }
+          return line;
+        })
+        .firstWhere((String line) => line.startsWith('DevTools listening'), orElse: () {
+          if (hitGlibcBug) {
+            _logger.printStatus(
+              'Encountered glibc bug https://sourceware.org/bugzilla/show_bug.cgi?id=19329. '
+              'Will try launching browser again.',
+            );
+            return null;
+          } else {
+            throw ToolExit('Failed to spawn: ${args.join(' ')}');
+          }
+        });
+
+      if (!hitGlibcBug) {
+        return process;
+      } else if (!exited) {
+        // A precaution that avoids accumulating browser processes, in case the
+        // glibc bug doesn't cause the browser to quit and we keep looping and
+        // launching more processes.
+        process.kill();
```

Comment:
Check if this is successful. What happens if the tool fails to kill it?

Reply:
How does the tool normally do this check? I can't find an example. I'm not sure what to do if it fails to kill it. I added it only as a precaution. I think it's OK if we fail. That would probably indicate that the system is in a completely busted and unrecoverable state.

Reply:
I don't have any advice :)

Reply:
Instead of using the `exited` flag, it's a little more idiomatic to put a timeout out `process.exitCode`, and then do `process.kill()` if it doesn't finish in time. Following sending the kill, if stdout might be useful for debugging, you'll want to explicitly drain that stream.